libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -232,3 +209,18 @@ TypePointer ReferencesResolver::typeFor(TypeName const& _typeName)
 	return _typeName.annotation().type = move(type);
 }
 
+void ReferencesResolver::typeError(string const& _description)
+{
+	auto err = make_shared<Error>(Error::Type::TypeError);
+	*err <<	errinfo_comment(_description);
```

Comment:
`ASTNode::createTypeError` added the source location of the node. Please also add that here, otherwise we will not get source locations for those errors.